Route/RoutePageGenerator.php

```diff
@@ -99,9 +99,20 @@ public function update(SiteInterface $site, OutputInterface $output = null)
         }
 
         // Iterate over declared routes from the routing mechanism
+        $routesDedupInfos = array();
+        // We first de-duplicate routes by path; the first declaration win !
         foreach ($this->router->getRouteCollection()->all() as $name => $route) {
-            $name = trim($name);
+            $routePath = $route->getPath();
+            if (array_key_exists($routePath, $routesDedupInfos)) {
+                $this->writeln($output, sprintf('  <info>IGNORE</info> route %s is ignored because it uses the same path as %s (path=%s)',
+                    $name, $routesDedupInfos[$routePath][0], $routePath));
```

Comment:
Please then correct the original files so it is obvious that we have to write this way ?

* https://github.com/sonata-project/SonataPageBundle/blob/3.x/Route/RoutePageGenerator.php#L73
* https://github.com/sonata-project/SonataPageBundle/blob/3.x/Route/RoutePageGenerator.php#L123
* https://github.com/sonata-project/SonataPageBundle/blob/3.x/Route/RoutePageGenerator.php#L155

You should run the php-cs-fixer you ask to use on the existing code. It would be easier for everyone ?